src/lib/parser/games/vanilla/config.ts

```diff
@@ -408,62 +409,103 @@ export class VanillaConfig implements GameConfig {
-          const extendedPartyArea = new Uint8Array(saveData.buffer, saveData.byteOffset + partyDataOffset + 32, 32)
-          for (let i = 0; i < 32; i++) {
-            if (extendedPartyArea[i] !== 0) {
-              return false // Has Quetzal patterns, reject
+        try {
+          const pokemon = this.createPokemonData(data)
+          if (pokemon.speciesId > 0) {
+            pokemonFound++
+
+            // Check for ROM hack features that vanilla shouldn't handle
+            if (pokemon.speciesId > 493) { // Extended species
+              hasRomHackFeatures = true
+              break
             }
-          }
 
-          // Check for extended Pokemon features if party is not empty
-          if (partyCount > 0) {
-            for (let slot = 0; slot < partyCount; slot++) {
-              const pokemonOffset = partyDataOffset + (slot * this.offsets.partyPokemonSize)
-              if (pokemonOffset + this.offsets.partyPokemonSize <= saveData.length) {
-                const pokemonData = new Uint8Array(saveData.buffer, saveData.byteOffset + pokemonOffset, this.offsets.partyPokemonSize)
-
-                // Check for extended species
-                const rawSpeciesId = new DataView(pokemonData.buffer, pokemonData.byteOffset + 0x20, 2).getUint16(0, true)
-                if (rawSpeciesId > 493) {
-                  return false // Extended species = Quetzal, reject
-                }
-
-                const pokemon = this.createPokemonData(pokemonData)
-
-                // Check for extended moves
-                const moves = [pokemon.move1, pokemon.move2, pokemon.move3, pokemon.move4]
-                if (moves.some(moveId => moveId > 354)) {
-                  return false // Extended moves = Quetzal, reject
-                }
-              }
+            // Check for radiant Pokemon (shiny number = 2)
+            const personality = new DataView(data.buffer, data.byteOffset).getUint32(0, true)
+            const shinyNumber = (personality >> 8) & 0xFF
+            if (shinyNumber === 2) { // Radiant Pokemon
+              hasRomHackFeatures = true
+              break
             }
+
+            // Check for extended moves
+            const moves = [pokemon.move1, pokemon.move2, pokemon.move3, pokemon.move4]
+            if (moves.some(moveId => moveId > 354)) { // Extended moves
+              hasRomHackFeatures = true
+              break
+            }
+          } else {
+            break // Empty slot, stop looking
           }
+        } catch {
+          break // Parsing failed, stop looking
         }
+      }
 
-        // If no Quetzal patterns found, this is a vanilla save
-        return true
+      // Reject saves with ROM hack features
+      if (hasRomHackFeatures) {
+        return false
       }
+
+      // If we found non-zero data but couldn't parse any Pokemon, this is likely a ROM hack save
+      if (hasNonZeroData && pokemonFound === 0) {
+        return false
+      }
```

Comment:
lets simplify this whole canHandle function to only be pokemonFound === 0 in the most simple and bare bones way